server/jobs/store2localinterface/store2localinterface.go

```diff
@@ -0,0 +1,289 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package store2localinterface
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+	"sync/atomic"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/kpng/server/jobs/kube2store"
+
+	"sigs.k8s.io/kpng/api/localnetv1"
+	"sigs.k8s.io/kpng/client/diffstore"
+	"sigs.k8s.io/kpng/server/pkg/endpoints"
+	"sigs.k8s.io/kpng/server/pkg/proxy"
+	"sigs.k8s.io/kpng/server/proxystore"
+	"sigs.k8s.io/kpng/server/serde"
+)
+
+type SendNodeLocalState struct {
+	consumer      NodeLocalStateConsumer
+	nodeName      string
+	ServicesDiff  *diffstore.Store[string, *diffstore.JSONLeaf[*localnetv1.Service]]
+	EndpointsDiff *diffstore.Store[string, *diffstore.JSONLeaf[*localnetv1.EndpointInfo]]
+	Store         *proxystore.Store
+}
+
+type NodeLocalStateConsumer interface {
+	UpdateServices(services <-chan *localnetv1.Service)
```

Comment:
what about just 1 stream with the Op object but made for internal use, like:
```golang
type Op struct {
  Sync bool // when it's a sync signal
  
  Set localnetv1.Set
  Path string
  Delete bool
  
  // 1st option:
  Value any
  // 2nd option
  Service *localnetv1.Service
  Endpoint *localnetv1.EndpointInfo
}
```

Reply:
also, if we're speaking internal / same memory space, is forcing a diff the way? We could nearly give the lightdiffstore instances, or maybe just a fullstate-like primary interface that could then be fed to the lightdiffstore in watchstate / the Local gRPC API

Reply:
> 1 stream with the Op object 

Are you referring to an [unidirectional GRPC stream](https://grpc.io/docs/what-is-grpc/core-concepts/#server-streaming-rpc) here or just the golang struct? Sorry getting overloaded with terms :) 

I'm fine with anything on the design of the transfer interface (struct interface etc) and just took the inspiration from Tim's Slides and [this ancient comment](https://github.com/kubernetes/enhancements/pull/2094#issuecomment-757063785) from the KEP lol

My end goal here is that it's super simple for folks to write backends, with or without any GRPC knowledge/familiarity. 

> also, if we're speaking internal / same memory space, is forcing a diff the way?

I'm not sure :) from an ease of use if we force the diff, or add the option to, then it takes the overhead of having to learn how to use a new library off of the user which IMO is a good thing? 

and if we're diffing ENLS I don't think it matters whether we do it before or after passing data to the user, from a scale concern at least. 

> We could nearly give the lightdiffstore instances, or maybe just a fullstate-like primary interface that could then be fed to the lightdiffstore in watchstate / the Local gRPC API

Totally... I'm not against that either, however I think collapsing some of the modules(watchstate/fullstate) into a single one wouldn't be the worse thing? It took me quite some time to dig in and understand all the various pieces modules (and where they live)